pollux/sql-doobie/src/main/scala/org/hyperledger/identus/pollux/sql/repository/JdbcCredentialSchemaRepository.scala

```diff
@@ -17,57 +17,64 @@ import java.util.UUID
 case class JdbcCredentialSchemaRepository(xa: Transactor[ContextAwareTask], xb: Transactor[Task])
     extends CredentialSchemaRepository {
   import CredentialSchemaSql.*
-  override def create(cs: CredentialSchema): RIO[WalletAccessContext, CredentialSchema] = {
+  override def create(cs: CredentialSchema): URIO[WalletAccessContext, CredentialSchema] = {
     ZIO.serviceWithZIO[WalletAccessContext](ctx =>
       CredentialSchemaSql
         .insert(CredentialSchemaRow.fromModel(cs, ctx.walletId))
         .transactWallet(xa)
+        .orDie
         .map(CredentialSchemaRow.toModel)
     )
   }
 
-  override def getByGuid(guid: UUID): Task[Option[CredentialSchema]] = {
+  override def findByGuid(guid: UUID): UIO[Option[CredentialSchema]] = {
     CredentialSchemaSql
       .findByGUID(guid)
       .transact(xb)
+      .orDie
       .map(
         _.headOption
           .map(CredentialSchemaRow.toModel)
       )
   }
 
-  override def update(cs: CredentialSchema): RIO[WalletAccessContext, Option[CredentialSchema]] = {
+  override def update(cs: CredentialSchema): URIO[WalletAccessContext, Option[CredentialSchema]] = {
```

Comment:
It should not return an `Option` if the record doesn't exist but rather throw a defect. If needed, it's up to the service making the call to check record exists (see next comment on `delete` method).

Reply:
fixed